### PR TITLE
Add support for shipping performance analyzer with RCA

### PR DESCRIPTION
**Pull request by ktkrg**
This small change adds support for packaging performance analyzer + RCA in ODFE.

Summary of changes:
* Separates the agent from the plugin location after install as the agent(with RCA) and the plugin are bundled together.
* Updates path to PA's conf files in docker entrypoint.

### CODE REVIEW

File: elasticsearch/docker/build/elasticsearch/bin/docker-entrypoint.sh [hunk 1]
```diff
@@ -90,8 +90,14 @@ fi
 
 if [[ -d "/usr/share/elasticsearch/plugins/opendistro_performance_analyzer" ]]; then
     CLK_TCK=`/usr/bin/getconf CLK_TCK`
-    ES_JAVA_OPTS="-Djava.security.policy=file:///usr/share/elasticsearch/plugins/opendistro_performance_analyzer/pa_config/es_security.policy -Dclk.tck=$CLK_TCK -Djdk.attach.allowAttachSelf=true $ES_JAVA_OPTS"
-    /usr/bin/supervisord -c /usr/share/elasticsearch/plugins/opendistro_performance_analyzer/pa_config/supervisord.conf
+    ES_JAVA_OPTS="-Dclk.tck=$CLK_TCK -Djdk.attach.allowAttachSelf=true $ES_JAVA_OPTS"
+    if [[ -d "/usr/share/elasticsearhc/performance-analyzer-rca" ]]; then
```
Comment (weicongs-amazon) [inline elasticsearch/docker/build/elasticsearch/bin/docker-entrypoint.sh:94]: typo: "elasticsearhc"

File: elasticsearch/docker/templates/Dockerfile.j2 [hunk 2]
```diff
@@ -108,6 +108,8 @@ RUN for plugin_path in `cat /tmp/plugins/plugins.list`; do \
       elasticsearch-plugin install --batch file:/tmp/plugins/$plugin_path; \
     done
 
+# Pull performance analyzer agent(RCA) from the plugin directory out into ESHOME
+RUN if [ -d plugins/opendistro_performance_analyzer/performance-analyzer-rca ]; then mv plugins/opendistro_performance_analyzer/performance-analyzer-rca ./; fi
```
Comment (weicongs-amazon) [inline elasticsearch/docker/templates/Dockerfile.j2:112]: why do we need this "move"?  could you use the same way as PA does?

File: elasticsearch/docker/templates/Dockerfile.j2 [hunk 3]
```diff
@@ -191,7 +193,7 @@ RUN chgrp 0 /usr/local/bin/docker-entrypoint.sh && \
     chmod g=u /etc/passwd && \
     chmod 0775 /usr/local/bin/docker-entrypoint.sh
 
-EXPOSE 9200 9300 9600
+EXPOSE 9200 9300 9600 9650
```
Comment (weicongs-amazon) [inline elasticsearch/docker/templates/Dockerfile.j2:196]: In the discussion meeting, you guys mentioned there is no new exposed port. what's this new one for?

File: elasticsearch/docker/templates/Dockerfile.j2 (hunk 2 above)
Comment (ktkrg) [reply elasticsearch/docker/templates/Dockerfile.j2:112]: The move is done by post-install hooks that DEB and RPM packages expose. There is no equivalent in docker world(where we install the plugin using zip) but to do that manually.

File: elasticsearch/docker/templates/Dockerfile.j2 [hunk 4]
```diff
@@ -108,8 +108,10 @@ RUN for plugin_path in `cat /tmp/plugins/plugins.list`; do \
       elasticsearch-plugin install --batch file:/tmp/plugins/$plugin_path; \
     done
 
-# Pull performance analyzer agent(RCA) from the plugin directory out into ESHOME
-RUN if [ -d plugins/opendistro_performance_analyzer/performance-analyzer-rca ]; then mv plugins/opendistro_performance_analyzer/performance-analyzer-rca ./; fi
+# Pull performance analyzer agent(RCA) from the plugin directory out into ESHOME and make the agent executable.
+RUN if [ -d plugins/opendistro_performance_analyzer/performance-analyzer-rca ]; then \
+     mv plugins/opendistro_performance_analyzer/performance-analyzer-rca ./ && \
+     chmod 755 performance-analyzer-rca/pa_bin/performance-analyzer-agent; fi
```
Comment (peterzhuamazon) [inline elasticsearch/docker/templates/Dockerfile.j2:114]: If we start to use performance-analyzer-cli for all distros, do we still need to change permission on agent?

Comment (ktkrg) [reply elasticsearch/docker/templates/Dockerfile.j2:114]: I tried to change references to performance-analyzer-agent to point to performance-analyzer-cli, but we seemed to need a little more work to get it done correctly.

I created an [issue](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/issues/263) in our RCA repository to consolidate them. We hope to address this by the next release.

**Comment by weicongs-amazon** [comment]
btw, what's the reason why this file "${PACKAGE_NAME}-${OD_VERSION}/data/rca_enabled.conf" is needed in the tarball, but not in docker image?

File: elasticsearch/docker/templates/Dockerfile.j2 (hunk 4 above)
Comment (peterzhuamazon) [reply elasticsearch/docker/templates/Dockerfile.j2:114]: Sure.

**Comment by ktkrg** [comment]
> btw, what's the reason why this file "${PACKAGE_NAME}-${OD_VERSION}/data/rca_enabled.conf" is needed in the tarball, but not in docker image?

It's not needed actually. It'll be created at run time if it doesn't exist.

**Pull request merged**